ktor-client/ktor-client-java/jvm/src/io/ktor/client/engine/java/JavaHttpEngine.kt

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.engine.java
+
+import io.ktor.client.engine.*
+import io.ktor.client.features.*
+import io.ktor.client.request.*
+import io.ktor.client.utils.*
+import io.ktor.util.*
+import kotlinx.coroutines.*
+import java.io.*
+import java.net.*
+import java.net.http.*
+import java.time.*
+import java.util.concurrent.*
+import kotlin.coroutines.*
+
+@InternalAPI
+@Suppress("KDocMissingDocumentation")
+public class JavaHttpEngine(override val config: JavaHttpConfig) : HttpClientEngineBase("ktor-java") {
+
+    public override val dispatcher: CoroutineDispatcher by lazy {
+        Dispatchers.clientDispatcher(
+            config.threadsCount,
+            "ktor-java-dispatcher"
+        )
+    }
+
+    public override val supportedCapabilities: Set<HttpTimeout.Feature> = setOf(HttpTimeout)
+
+    override val coroutineContext: CoroutineContext
+
+    private val requestsJob = SilentSupervisor(super.coroutineContext[Job])
+
+    private var httpClient: HttpClient? = null
+
+    /**
+     * Exposed for tests only.
+     */
+    internal val executor = Executors.newCachedThreadPool {
+        Thread(it, "ktor-client-java").apply {
```

Comment:
It would be nice to also have some `id` number here

Reply:
I've applied changes, now each new thread has `number` in suffux.